pkg/authorization/workspace_content_authorizer.go

```diff
@@ -98,14 +120,31 @@ func (a *workspaceContentAuthorizer) Authorize(ctx context.Context, attr authori
 		if isAuthenticated && (isUser || isServiceAccountFromRootCluster) {
 			withGroups := deepCopyAttributes(attr)
 			withGroups.User.(*user.DefaultInfo).Groups = append(attr.GetUser().GetGroups(), bootstrap.SystemKcpClusterWorkspaceAccessGroup)
+
+			kaudit.AddAuditAnnotations(
+				ctx,
+				WorkspaceContentAuditDecision, DecisionAllowed,
+				WorkspaceContentAuditReason, "subject is either an authenticated user or a serviceaccount in root",
+			)
+
 			return a.delegate.Authorize(ctx, withGroups)
 		}
+		kaudit.AddAuditAnnotations(
+			ctx,
+			WorkspaceContentAuditDecision, DecisionNoOpinion,
+			WorkspaceContentAuditReason, "only root serviceaccounts may access root",
```

Comment:
same as above, this is a possible resolution, the reason is rather something like `toplevel.authorization.kcp.io/reason: root workspace access by non-root service account not permitted`.